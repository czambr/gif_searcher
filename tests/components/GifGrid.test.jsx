import { render, screen } from '@testing-library/react';
import { GifGrid } from '../../src/components/GifGrid';
import { useFetchGifs } from '../../src/hooks/useFetchGifs';

jest.mock('../../src/hooks/useFetchGifs');

describe('Evaluando el componente <GifGrid />', () => {
    const category = 'Dragon Ball';

    test('Debe hacer Match con el snapshot cuando no existen datos', () => {
        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true,
        });
        const { container } = render(<GifGrid category={category} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar el loading inicialmente', () => {
        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true,
        });

        render(<GifGrid category={category} />);
        expect(screen.getByText('Cargando...'));
        expect(screen.getByText(category));
    });

    test('Debe de mostrar items cuando se cargan las imagenes de useFetchGifs()', () => {
        const gifs = [
            {
                id: 'abc',
                title: 'Dragon Ball',
                url: 'https://localhost:dragonball.jpg',
            },
            {
                id: '123',
                title: 'Avatar',
                url: 'https://localhost:avatar.jpg',
            },
        ];

        useFetchGifs.mockReturnValue({
            images: gifs,
            isLoading: false,
        });

        render(<GifGrid category={category} />);
        expect(screen.getAllByRole('img').length).toBe(2);
    });
});
