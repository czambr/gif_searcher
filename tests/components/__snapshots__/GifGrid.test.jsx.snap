// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evaluando el componente <GifGrid /> Debe hacer Match con el snapshot cuando no existen datos 1`] = `
<div>
  <h3>
    Dragon Ball
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
